### PR TITLE
[kube-state-metrics] fix/split Registry and Repository Values in a proper way

### LINKED ISSUE #3075: [kube-state-metrics] Image breaks when passing global.imageRegistry
### Describe the bug a clear and concise description of what the bug is.

The Deployment image is misconfigured when passing a global.imageRegistry value, and image.repository is completely ignored. 

So if I pass:
```
global:
  imageRegistry: my-private-ecr.aws.com # as an example
```

The result of the [deployment image](https://github.com/prometheus-community/helm-charts/blob/4c2cca2750499044801d82ce65bbe487d3c3f464/charts/kube-state-metrics/templates/deployment.yaml#L125) 
 would be: `my-private-ecr.aws.com:v2.8.0`. The expected output is `my-private-ecr.aws.com/custom/kube-state-metrics/kube-state-metrics:v2.8.0`.

As a working expected example you could refer to the [Daemonset image for prometheus-node-exporter](https://github.com/prometheus-community/helm-charts/blob/4c2cca2750499044801d82ce65bbe487d3c3f464/charts/prometheus-node-exporter/templates/daemonset.yaml#L45). This chart overrides the registry but not the repo.

In summary, it breaks also the kube-prometheus-stack chart, since if you use the latest version, it will use the latest version of the kube-state-metrics chart which breaks if you use the `global` values for custom imageRegistry.

### What's your helm version?

v3.11.0

### What's your kubectl version?

v1.25.4

### Which chart?

kube-state-metrics

### What's the chart version?

4.31.0

### What happened?

_No response_

### What you expected to happen?

That the deployment image respects the `global.imageRegistry` without breaking the deployment image.

### How to reproduce it?

Just run the helm template with the `global.imageRegistry` set to whatever, and then look at the `image` of the Deployment and you'll see that it's broken.

### Enter the changed values of values.yaml?

global:
  imageRegistry: my-private-ecr.aws.com # as an example

### Enter the command that you execute and failing/misfunctioning.

helm template kube-state-metrics prometheus-community/kube-state-metrics --version 4.31.0 --set global.imageRegistry=my-private-ecr.aws.com

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by brainfair**
#### What this PR does / why we need it

We need to split registry and repository values in a proper way as usual in another charts. Also currently it's a bug if you want to use the global imageRegistry variable in kube-prometheus-stack it actually broke this sub chart, because currently, we expect a full path for the image, but in all others, we expect here only the registry name.

#### Which issue this PR fixes

- fixes #3075 

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/kube-state-metrics/templates/_helpers.tpl [hunk 1]
```diff
@@ -116,3 +116,41 @@ Formats imagePullSecrets. Input is (dict "Values" .Values "imagePullSecrets" .{s
...
+{{- printf "%s/%s:%s@%s" .Values.kubeRBACProxy.image.registry .Values.kubeRBACProxy.image.repository (default (printf "v%s" .Chart.AppVersion) .Values.kubeRBACProxy.image.tag) .Values.kubeRBACProxy.image.sha }}
+{{- end }}
+{{- else }}
+{{- if .Values.global.imageRegistry }}
+{{- printf "%s/%s:%s" .Values.global.imageRegistry .Values.kubeRBACProxy.image.repository (default (printf "v%s" .Chart.AppVersion) .Values.kubeRBACProxy.image.tag) }}
+{{- else }}
+{{- printf "%s/%s:%s" .Values.kubeRBACProxy.image.registry .Values.kubeRBACProxy.image.repository (default (printf "v%s" .Chart.AppVersion) .Values.kubeRBACProxy.image.tag) }}
+{{- end }}
+{{- end }}
+{{- end }}
```
Comment (QuentinBisson) [inline charts/kube-state-metrics/templates/_helpers.tpl:156]:
```suggestion
{{- end }}

```

Comment (brainfair) [reply charts/kube-state-metrics/templates/_helpers.tpl]: ty, done

**Review by QuentinBisson — APPROVED** [review]
I'd love to merge but I do not have rights on this repo yet

**Comment by QuentinBisson** [comment]
@brainfair I think you need to bump the chart version

**Comment by brainfair** [comment]
@QuentinBisson yes sure, updated

**Comment by QuentinBisson** [comment]
@mrueg do you mind reviewing this one?

File: charts/kube-state-metrics/Chart.yaml [hunk 2]
```diff
@@ -7,7 +7,7 @@ keywords:
 - prometheus
 - kubernetes
 type: application
-version: 4.32.0
+version: 4.33.0
```
Comment (mrueg) [inline charts/kube-state-metrics/Chart.yaml:10]: This is a breaking change, so major should get bumped

**Review by dotdc — APPROVED** [review]
Sorry for the delay.
Didn't try with rbac proxy, but everything seems fine, so I think we can merge this now.

**Pull request merged**